lighthouse-core/audits/byte-efficiency/legacy-javascript.js

```diff
@@ -348,18 +347,89 @@ class LegacyJavascript extends Audit {
     return urlToMatchResults;
   }
 
+  /**
+   * @param {PatternMatchResult[]} matches
+   */
+  static estimateWastedBytes(matches) {
+    const polyfillSignals = matches.filter(m => !m.name.startsWith('@')).map(m => m.name);
+    const transformSignals = matches.filter(m => m.name.startsWith('@')).map(m => m.name);
+
+    // experimenting.
+    const POLYFILL_USE_GRAPH = true;
+
+    let estimatedWastedBytesFromPolyfills = 0;
+    if (POLYFILL_USE_GRAPH) {
+      // TODO
+      const graph = require('./polyfill-graph-data.json');
+      const modulesSeen = new Set();
+      for (const polyfillSignal of polyfillSignals) {
+        // @ts-ignore: lacking index type.
+        const modules = graph.dependencies[polyfillSignal];
+        for (const module of modules) {
+          modulesSeen.add(module);
+        }
+      }
+
+      if (polyfillSignals.length > 0) estimatedWastedBytesFromPolyfills += graph.baseSize;
+      estimatedWastedBytesFromPolyfills += graph.baseSize;
+      estimatedWastedBytesFromPolyfills += [...modulesSeen].reduce((acc, module) => {
+        return acc + graph.moduleSizes[module];
+      }, 0);
+      // Not needed?
+      estimatedWastedBytesFromPolyfills =
+        Math.min(estimatedWastedBytesFromPolyfills, graph.maxSize);
+    } else {
```

Comment:
I like the graph a lot more.

Reply:
yeah the implementation makes more sense the hard part was the graph creation which you've already done, so sg :)

just a pain to upkeep i'd say